src/foam/dao/index/AATree.js

```diff
@@ -358,7 +358,7 @@ foam.CLASS({
         return copy;
       }
 
-      if ( r < 0 ) return s.right.lte(key, compare, nullNode);
+      if ( r < 0 ) return s.left.lte(key, compare, nullNode);
```

Comment:
This looks like a "how did this ever work?" kind of thing. @marcogbarcellos can we get a test that would have failed with the old code?